criu/netfilter.c

```diff
@@ -297,9 +297,9 @@ int nftables_lock_connection(struct inet_sk_desc *sk)
 	return ret;
 }
 
-int nftables_get_table(char *table, int n)
+int nftables_get_table(char *table, int n, uint64_t id)
 {
-	if (snprintf(table, n, "inet CRIU-%d", root_item->pid->real) < 0) {
+	if (snprintf(table, n, "inet CRIU-%" PRIx64, id) < 0) {
```

Comment:
```
	/*
	 * Keep compatibility with images
	 * without he->dump_criu_run_id field.
	 */
	if (!id) {
		if (!(root_ns_mask & CLONE_NEWPID)) {
			id = root_item->pid->real;
		} else {
			pr_err("Cannot generate CRIU's nftables table name because of issue #2550\n");
			return -1;
		}
		
	}

	if (snprintf(table, n, "inet CRIU-%" PRIx64, id) < 0) {
```
What do you think about this?